tensorflow-tools/src/main/java/org/tensorflow/tools/ndarray/impl/sequence/FastElementSequence.java

```diff
@@ -0,0 +1,75 @@
+/*
+ *  Copyright 2019 The TensorFlow Authors. All Rights Reserved.
+ *
+ *  Licensed under the Apache License, Version 2.0 (the "License");
+ *  you may not use this file except in compliance with the License.
+ *  You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *  Unless required by applicable law or agreed to in writing, software
+ *  distributed under the License is distributed on an "AS IS" BASIS,
+ *  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *  See the License for the specific language governing permissions and
+ *  limitations under the License.
+ *  =======================================================================
+ */
+
+package org.tensorflow.tools.ndarray.impl.sequence;
+
+import java.util.Iterator;
+import java.util.function.BiConsumer;
+
+import org.tensorflow.tools.buffer.DataBufferWindow;
+import org.tensorflow.tools.ndarray.NdArray;
+import org.tensorflow.tools.ndarray.NdArraySequence;
+import org.tensorflow.tools.ndarray.impl.AbstractNdArray;
+
+public final class FastElementSequence<T, U extends NdArray<T>> implements NdArraySequence<U> {
```

Comment:
This is an internal class (under `impl` package) and a lot of them are not documented. I can add something for this one now but probably it would be worth adding doc to all other internal classes as well.